src/Temporalio/Worker/Tuning/ICustomSlotSupplier.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Temporalio.Worker.Tuning
+{
+    /// <summary>
+    /// This class can be implemented to provide custom slot supplier behavior.
+    /// </summary>
+    /// <remarks>
+    /// WARNING: Custom slot suppliers are currently experimental.
+    /// </remarks>
+    public interface ICustomSlotSupplier : ISlotSupplier
```

Comment:
Could be discussed whether this is better as an abstract class, but I am fine with this too